File: backend/src/main/java/com/andrew/hdss/dtos/IndividualDto.java
package com.andrew.hdss.dtos;

import com.andrew.hdss.models.Individual;
import com.andrew.hdss.models.enums.Sex;

import java.time.LocalDate;

public record IndividualDto(
        Long id,
        String clientId,
        String extendedId,
        String firstName,
        String lastName,
        Sex sex,
        LocalDate dateOfBirth,
        boolean dobEstimated,
        Long motherId,
        Long fatherId
) {
    public static IndividualDto from(Individual i) {
        return new IndividualDto(
                i.getId(),
                i.getClientId(),
                i.getExtendedId(),
                i.getFirstName(),
                i.getLastName(),
                i.getSex(),
                i.getDateOfBirth(),
                i.isDobEstimated(),
                i.getMother() != null ? i.getMother().getId() : null,
                i.getFather() != null ? i.getFather().getId() : null
        );
    }
}

File: android/app/src/main/java/com/andrew/hdss/dtos/IndividualDto.kt
package com.andrew.hdss.dtos

import com.andrew.hdss.data.models.enums.Sex
import kotlinx.serialization.Serializable

@Serializable
data class IndividualDto(
    val id: Long,
    val clientId: String,
    val extendedId: String,
    val firstName: String,
    val lastName: String,
    val sex: Sex,
    val dateOfBirth: String,
    val dobEstimated: Boolean,
    val motherId: Long?,
    val fatherId: Long?
)
